axelrod/player.py

```diff
@@ -258,6 +258,64 @@ def reset(self):
     def update_history(self, play, coplay):
         self.history.append(play, coplay)
 
+    def check_assumptions(self, game_characteristics: dict, raise_error: bool=True):
```

Comment:
Should the assumptions check be here or in the `Match` class? It's in a `Match` that `Player`s and `Game`s are combined (and also in the Moran process class, which uses the `Match` class). The `Player` class just needs to be initialized properly and be able to return values when `.strategy()` is called, it's not involved in the scoring process and so it doesn't need to care about the `Game` usually (unless it requires info about the game). Note that [it's in](https://github.com/Axelrod-Python/Axelrod/blob/dev/axelrod/match.py#L115) a `Match` that `Player` instances are currently given info about the game or match.

Reply:
checking assumptions has use cases outside of matches - if it's in the Player class we can use it for filtering, e.g. if we wanted all strategies from a list `strategies` that work on a game with assumptions `my_assumptions`, we could do

`[strategy() for strategy in strategies if strategy().assumptions_satisfy(my_assumptions)]`.

Note that in general, the methods just take a dict, not a `Game` object specifically; the `Game` itself is still completely hidden from the `Player` class. [In the Match class, the `Player`s and `Game` are combined in the way that you want, without either of them having their whole class exposed to the other.](https://github.com/Axelrod-Python/Axelrod/pull/1418/files#diff-78d3d04c3d62ef794070aad3a16585106965b56363df56323b5854ff1f2dfa64R119-R126)

> it doesn't need to care about the Game usually (unless it requires info about the game).

"how should strategies handle requiring info about the game" is really the issue that we want to solve in this PR! :)